### PR TITLE
[TACHYON-1007][TACHYON-984] Reenable raw table master client related API and tests

### PR DESCRIPTION
**Pull request by uronce-cc**
https://tachyon.atlassian.net/browse/TACHYON-984

Also, I tried to fix two TODOs left by gene and jiri on making a utility to remove the redundant retry logic because I need to write a RawTableMasterClient and I don't like the redundant codes.

### CODE REVIEW

File: common/src/main/java/tachyon/client/FileSystemMasterClient.java [hunk 1]
```diff
@@ -79,20 +77,21 @@ protected void afterConnect() {
...
+        public Long call() throws PropagateTException, TException {
+          try {
+            return mClient.getFileId(path);
+          } catch (InvalidPathException e) {
+            throw new PropagateTException(e);
+          }
+        }
+      });
+    } catch (RpcCallableWithPropagateTException.PropagateTException e) {
+      throw (InvalidPathException) e.getWrappedTException();
```
Comment (uronce-cc) [inline common/src/main/java/tachyon/client/FileSystemMasterClient.java]: Could you find a better way to propagate exceptions from `call()`?

Comment (jsimsa) [inline common/src/main/java/tachyon/client/FileSystemMasterClient.java]: I don't think that the InvalidPathException has to be caught since it is simply rethrown.

Comment (jsimsa) [inline common/src/main/java/tachyon/client/FileSystemMasterClient.java]: On a more general note, I suggest you separate your RPC retry logic into a separate PR so that we can merge the raw table changes independent of refining the RPC retry logic.

**Comment by jsimsa** [comment]
I suggest you split this PR into two. One for the RawTable logic and one for the RPC retry logic.

I am not sure if the current approach to factoring out the RPC retry logic is good enough as it sometimes produces more boilerplate than the original code. I think we could use reflection to accomplish the same. Let me draft a reflection-based solution for you to take a look at.

**Comment by uronce-cc** [comment]
@jsimsa thanks, I'll open another PR to enable raw table client only.

 For the retry, I tried another solution by passing the class type of the exceptions that need to be rethrown, and filtering the `TException` to see whether it is of the specified class type, if it is, rethrow it, but since I can not specify different exceptions in the signature of `call()`, they have to be casted in the `try catch` block around `retryRpc` too. It will be great for a solution to remove the boilerplate.
